src/MarkdownHeading.js

```diff
@@ -17,7 +17,7 @@ const StyledHeading = styled(Heading)`
   }
 `
 const MarkdownHeading = ({children, className, ...rest}) => {
-  const id = slugify(children.toString())
+  const id = slugify(children.toString(), {lower: true})
```

Comment:
Hmm no, I wonder what would happen if the child was a component 🤔 I'll test it out

Reply:
![image](https://user-images.githubusercontent.com/8960591/57153339-e0721c80-6d8a-11e9-820d-4683e99bce74.png)
😬

Reply:
I'm not really sure if it's worth putting in provisions for edge cases in the code, we can just add to the documentation that the children should be text only. Maybe add a prop type declaration for `children`?

Reply:
Especially since this is to be used in MD or MDX files only, the only way you'd get a non-string in there is if you pass in a React component which I feel like is fair to say we should do 😬 Not sure if Markdown will even convert `### <Text>asklfjasdf</Text>` to a heading anyways

Reply:
Added in a check to make sure children isn't empty!